src/utils/db/db.ts

```diff
@@ -496,6 +496,49 @@ export class DB {
         return person
     }
 
+    public async setPersonAsIdentified(person: Person): Promise<void> {
+        let shouldUpdateClickhouse = false
+
+        const result = await this.postgresQuery(
+            `
+                UPDATE posthog_person 
+                SET is_identified=$1 
+                WHERE id = $2 
+                AND is_identified = $3
+                RETURNING is_identified
+            `,
+            [true, person.id, false],
+            'setPersonAsIdentified'
+        )
+
+        if (result.rows.length > 0) {
+            shouldUpdateClickhouse = true
+        }
+
+        if (this.kafkaProducer && shouldUpdateClickhouse) {
+            const message = {
+                topic: KAFKA_PERSON,
+                messages: [
+                    {
+                        value: Buffer.from(
+                            JSON.stringify({
+                                created_at: castTimestampOrNow(
+                                    person.created_at,
+                                    TimestampFormat.ClickHouseSecondPrecision
+                                ),
+                                properties: JSON.stringify(person.properties),
+                                team_id: person.team_id,
+                                is_identified: true,
+                                id: person.uuid,
+                            })
+                        ),
+                    },
+                ],
+            }
```

Comment:
I assume you'll deal with this duplication as you 
> This will also make even more sense once I completely re-do updatePerson, as I want to have that focus solely on properties.